lib/workers/repository/onboarding/pr/index.ts

```diff
@@ -152,14 +152,16 @@ If you need any further assistance then you can also [request help here](${confi
       logger.info({ pr: pr.displayNumber }, 'Onboarding PR created');
       await addParticipants(config, pr);
     }
-  } catch (err) /* istanbul ignore next */ {
+  } catch (err) {
     if (
       err.statusCode === 422 &&
-      err.response?.body?.errors?.[0]?.message?.startsWith(
+      err.body?.errors?.[0]?.message?.startsWith(
```

Comment:
Tested this change in a real repo?

Reply:
I did not but i will, ive changed it mainly because of this:
https://github.com/renovatebot/renovate/blob/1b28501d30fefa502235b886518133f998db4ea0/lib/util/http/github.ts#L131-L145
but also i couldnt find a proper type to handle it while writing the tests for it.

Reply:
i think the other code is wrong, it should be a request error object of got, which has the response property. the direct body property is legacy. checkout `util/http/legacy.ts`

Reply:
actually they both work because of `util/http/legacy.ts`, the `err.[statusCode|body]` exists on the prototype chain as it set in `legacy.ts`. while `response.body` is an own property. 

so because of `// TODO: remove when code is refactored (#9651)` from `legacy` we should probably use the response field.

as for the tests, im currently using `GotLegacyError`
guess ill have to change it to `HTTPError` from got
![image](https://user-images.githubusercontent.com/97394622/170669638-98652595-bf09-41ce-b022-8e8b5985213c.png)